_posts/2025-01-15-moving-data-between-JBOD-disks-using-cruise-control.md

```diff
@@ -0,0 +1,559 @@
+---
+layout: post
+title: "Moving data between JBOD disks using Cruise Control"
+date: 2025-01-15
+author: shubham_rawat
+---
+
+Apache Kafka is a platform that provides durability and fault tolerance by storing messages on persistent volumes.
+In most cases, each Kafka broker will use one persistent volume.
+However, it is also possible to use multiple volumes for each broker.
+This configuration is called JBOD storage.
+Using JBOD storage, you can increase the data storage capacity for Kafka nodes, which can further lead to performance improvements.
+It might happen that you need to add or remove volumes to increase or shrink the overall capacity and performance of the Kafka cluster.
+When adding new volumes, you first need to add the volume and then move some data to it.
+That can be done using the [_intrabroker_](https://strimzi.io/docs/operators/in-development/deploying#con-rebalance-str) rebalance.
+When removing volumes, you have to first safely move the data to other volumes first.
+Failing to do so could result in data loss.
+Moving data between the JBOD disks can be done using the `kafka-reassign-partitions.sh` tool, which is not very user-friendly, therefore in Strimzi 0.45.0 we introduced the ability to move data between the JBOD disks using Cruise Control.
```

Comment:
Maybe check with @PaulRMellor ... but I wonder if someting like this would sound better.

```suggestion
Moving data between the JBOD disks can be done using the `kafka-reassign-partitions.sh` tool, which is not very user-friendly.
Therefore - in Strimzi 0.45.0 - we introduced the ability to move data between the JBOD disks using Cruise Control.
```

Reply:
Looks good